lightning/src/ln/reorg_tests.rs

```diff
@@ -0,0 +1,154 @@
+		let node_1_commitment_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
+		assert_eq!(node_1_commitment_txn.len(), 3); // ChannelMonitor: 1 offered HTLC-Timeout, ChannelManger: 1 local commitment tx, 1 Offered HTLC-Timeout
+		assert_eq!(node_1_commitment_txn[1].output.len(), 2); // to-local and Offered HTLC (to-remote is dust)
+		check_spends!(node_1_commitment_txn[1], chan_2.3);
+		check_spends!(node_1_commitment_txn[2], node_1_commitment_txn[1].clone());
+		check_spends!(node_1_commitment_txn[0], node_2_commitment_txn[0]);
+
+		// Confirm node 2's commitment txn (and node 1's HTLC-Timeout) on node 1
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: vec![node_2_commitment_txn[0].clone(), node_1_commitment_txn[0].clone()] }, CHAN_CONFIRM_DEPTH + 1);
+		// ...but return node 2's commitment tx (and claim) in case claim is set and we're preparing to reorg
+		node_2_commitment_txn
+	};
+	check_closed_broadcast!(nodes[1], false); // We should get a BroadcastChannelUpdate (and *only* a BroadcstChannelUpdate)
+	headers.push(header.clone());
+	// At CHAN_CONFIRM_DEPTH + 1 we have a confirmation count of 1, so CHAN_CONFIRM_DEPTH +
+	// ANTI_REORG_DELAY - 1 will give us a confirmation count of ANTI_REORG_DELAY - 1.
+	for i in CHAN_CONFIRM_DEPTH + 2..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1 {
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.bitcoin_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected_checked(&header, i, &vec![], &[0; 0]);
+		headers.push(header.clone());
+	}
+	check_added_monitors!(nodes[1], 0);
+	assert_eq!(nodes[1].node.get_and_clear_pending_events().len(), 0);
+
+	if claim {
+		// Now reorg back to CHAN_CONFIRM_DEPTH and confirm node 2's broadcasted transactions:
+		for (height, header) in (CHAN_CONFIRM_DEPTH + 1..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1).zip(headers.iter()).rev() {
+			nodes[1].block_notifier.block_disconnected(&header, height);
+		}
+
+		header = BlockHeader { version: 0x20000000, prev_blockhash: Default::default(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: claim_txn }, CHAN_CONFIRM_DEPTH + 1);
+
+		// ChannelManager only polls ManyChannelMonitor::get_and_clear_pending_htlcs_updated when we
+		// probe it for events, so we probe non-message events here (which should still end up empty):
+		assert_eq!(nodes[1].node.get_and_clear_pending_events().len(), 0);
+	} else {
+		// Confirm the timeout tx and check that we fail the HTLC backwards
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.bitcoin_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected_checked(&header, CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY, &vec![], &[0; 0]);
```

Comment:
Why not connect ANTI_REORG_DELAY blocks here?

Reply:
We already connected ANTI_REORG_DELAY - 1 blocks above, this is just the last one to get it to ANTI_REORG_DELAY.